### PR TITLE
feat: issuer profile

### PR DESCRIPTION
**Pull request by vkubiv**
Signed-off-by: Volodymyr Kubiv <volodymyr.kubiv@euristiq.com>

### CODE REVIEW

**Review by aholovko — APPROVED** [review]

File: pkg/did/creator.go [hunk 1]
```diff
@@ -17,37 +17,57 @@ import (
...
 // Config configures PublicDID.
 type Config struct {
-	Method                 string
-	VerificationMethodType string
-	VDR                    vdr.Registry
-	JWKKeyCreator          func(kms.KeyManager) (string, *jwk.JWK, error)
-	CryptoKeyCreator       func(kms.KeyManager) (string, interface{}, error)
-	DIDAnchorOrigin        string
+	VDR             vdr.Registry
+	DIDAnchorOrigin string
```
Comment (sudeshrshetty) [inline pkg/did/creator.go:41]: this one is needed for orb only right?

File: pkg/doc/vc/validate.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	jwtVC = "jwt_vc"
+	ldpVC = "ldp_vc"
+)
+
+/*
+"*/
+
+func ValidateVCFormat(format string) error {
+	if format != jwtVC && format != ldpVC {
+		return fmt.Errorf("unsupported vc format %s", format)
```
Comment (sudeshrshetty) [inline pkg/doc/vc/validate.go:26]: also show supported vc formats in error message

File: pkg/doc/vc/validate.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+		}
+	}
+
+	if format == ldpVC {
+		validSignatureTypes = []string{
+			"Ed25519Signature2018",
+			"Ed25519Signature2020",
+			"EcdsaSecp256k1Signature2019",
+			"BbsBlsSignature2020",
+			"JsonWebSignature2020",
```
Comment (sudeshrshetty) [inline pkg/doc/vc/validate.go:49]: can we keep these in constants?
something like this https://github.com/hyperledger/aries-framework-go/blob/main/pkg/wallet/contents.go#L31-L73

And try to make checks case insensitive

File: pkg/did/creator.go [hunk 4]
```diff
@@ -17,37 +17,57 @@ import (
...
-		method, supported := methods[config.Method]
-		if !supported {
-			return nil, fmt.Errorf("unsupported did method: %s", config.Method)
-		}
+// PublicDID creates a new public DID given a key manager.
+func (c *Creator) PublicDID(method, verificationMethodType string, keyType kms.KeyType,
+	km KeysCreator) (*CreateResult, error) {
+	methods := map[string]func(verificationMethodType string, keyType kms.KeyType, km KeysCreator) (*CreateResult, error){
+		key.DIDMethod: c.keyDID,
+		orb.DIDMethod: c.createDID,
```
Comment (sudeshrshetty) [inline pkg/did/creator.go:56]: also add did web here, but throw error in create func for now

Comment (sudeshrshetty) [reply pkg/did/creator.go:56]: or add a TODO to add `did:web` support and create an issue.

File: pkg/doc/vc/validate.go (hunk 3 above)
Comment (sudeshrshetty) [reply pkg/doc/vc/validate.go:49]: objective: should be easier to add new key types in future.

File: pkg/doc/vc/validate.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	if validSignatureTypes == nil {
+		return "", fmt.Errorf("unsupported vc format %s", format)
+	}
+
+	for _, val := range validSignatureTypes {
+		if val.lowerCase() == strings.ToLower(signatureType) {
+			return val, nil
+		}
+	}
+
```
Comment (sudeshrshetty) [inline pkg/doc/vc/validate.go:90]: Can we have something like `signatuteType.isValid(format)`

Similar to https://github.com/hyperledger/aries-framework-go/blob/main/pkg/wallet/contents.go#L60-L68

Comment (sudeshrshetty) [inline pkg/doc/vc/validate.go:90]: this whole can be replaced by something like this `signatureType.isValid(format)`


Similar to https://github.com/hyperledger/aries-framework-go/blob/main/pkg/wallet/contents.go#L60-L68

Comment (sudeshrshetty) [reply pkg/doc/vc/validate.go:90]: @vkubiv as discussed, please fix it in next PR

**Pull request merged**